File: 9_Encapsulation.py
class Car :
    
    def __init__(self):
        self.__speed =0
        
    def set_speed(self, speed):
        if speed > 0 and speed < 200:
            self.__speed = speed
        else:
            print("Invalid speed")
        
        
    def get_speed(self):
        return self.__speed
    
car1=Car()
car1.set_speed(20)
print(car1.get_speed())
        

File: 9_Encapsulation.cpp


#include <bits/stdc++.h>
using namespace std;
#define ll long long

class Car
{
private:
    int speed;

public:
    void setSpeed(int speed)
    {
        if (speed > 0)
            this->speed = speed;
    }
    int getSpeed()
    {
        return speed;
    }
};
int main()
{
    Car car1;
    car1.setSpeed(60);
    cout << "speed : " << car1.getSpeed() << endl;

    return 0;
}